latch_cli/utils.py

```diff
@@ -138,6 +140,67 @@ def hash_directory(dir_path: Path) -> str:
     return m.hexdigest()
 
 
+def generate_temporary_ssh_credentials(ssh_key_path: Path) -> str:
+    # check if there is already a valid key at that path, and if so, use that
+    # otherwise, if its not valid, remove it
+    if ssh_key_path.exists():
+        try:
+            # check if file is valid + print out a fingerprint for the key
+            cmd = ["ssh-keygen", "-l", "-f", ssh_key_path]
+            valid_private_key = subprocess.run(cmd, check=True)
+            cmd = ["ssh-keygen", "-l", "-f", ssh_key_path.with_suffix(".pub")]
+            valid_public_key = subprocess.run(cmd, check=True)
+            if valid_private_key.stdout != valid_public_key.stdout:
+                raise
+            else:
+                # if both files are valid and their fingerprints match, use them instead of generating a new pair
+                print(f"Found existing key pair at {ssh_key_path}.")
+        except:
+            print(f"Found malformed key-pair at {ssh_key_path}. Overwriting.")
+            ssh_key_path.unlink(missing_ok=True)
+            ssh_key_path.with_suffix(".pub").unlink(missing_ok=True)
+
+    # generate private key
+    if not ssh_key_path.exists():
+        cmd = ["ssh-keygen", "-f", ssh_key_path, "-N", "", "-q"]
+        try:
+            subprocess.run(cmd, check=True)
+        except subprocess.CalledProcessError as e:
+            raise ValueError(
+                "There was a problem creating temporary SSH credentials. Please ensure that "
+                "`ssh-keygen` is installed and available in your PATH."
+            )
+        os.chmod(ssh_key_path, int("700", base=8))
+
+    # make key available to ssh-agent daemon
+    cmd = ["ssh-add", ssh_key_path]
+    try:
+        subprocess.run(cmd, check=True)
+    except subprocess.CalledProcessError as e:
+        raise ValueError(
+            "There was an issue adding temporary SSH credentials to your SSH Agent. Please ensure "
+            "that your SSH Agent is running, or (re)start it manually by running\n\n    $ eval `ssh-agent -s`"
+            "\n\n"
+        )
+
+    # decode private key into public key
+    cmd = ["ssh-keygen", "-y", "-f", ssh_key_path]
+    try:
+        out = subprocess.run(cmd, check=True, capture_output=True)
+    except subprocess.CalledProcessError as e:
+        cmd = ["cat", ssh_key_path.with_suffix(".pub")]
```

Comment:
how is this a safe fallback, what if the pub key is not there, what if the pub key that is there is for some reason not associated with the private key

Reply:
as in the file name alone matches, i imagine this especially common with `id_rsa` named keys

Reply:
fair yeah, will del this